File: Trackify_Admin/inject_fake_error.js
import { initializeApp } from 'firebase/app';
import { getFirestore, collection, addDoc, serverTimestamp } from 'firebase/firestore';

const firebaseConfig = {
  apiKey: "AIzaSyDC9fZJTOi455NsbN2AQRnGz_cjZvmwedc",
  authDomain: "ys-construction.firebaseapp.com",
  projectId: "ys-construction",
  storageBucket: "ys-construction.firebasestorage.app",
  messagingSenderId: "487752590406",
  appId: "1:487752590406:web:cd934161591117bc581647"
};

const app = initializeApp(firebaseConfig);
const db = getFirestore(app);

async function inject() {
  await addDoc(collection(db, 'error_logs'), {
    message: '[FAKE ERROR] Connection to Main Tracking Node Lost.',
    stackTrace: 'Exception: SocketTimeout\\n  at TrackingService.sync (tracking_service.dart:45)',
    type: 'Network Error',
    severity: 'HIGH',
    status: 'NEW',
    module: 'Tracking Node',
    createdAt: serverTimestamp(),
    userId: 'admin_test'
  });
  console.log("Injected fake error!");
  process.exit(0);
}

inject();

File: tools/data-fixes/inject_fake_error.py
import firebase_admin
from firebase_admin import credentials
from firebase_admin import firestore

try:
    cred = credentials.Certificate('Trackify_Admin/serviceAccountKey.json')
    firebase_admin.initialize_app(cred)
except ValueError:
    pass

db = firestore.client()

db.collection('error_logs').add({
    'message': '[FAKE ERROR] Connection to Main Tracking Node Lost.',
    'stackTrace': 'Exception: SocketTimeout\\n  at TrackingService.sync (tracking_service.dart:45)',
    'type': 'Network Error',
    'severity': 'HIGH',
    'status': 'NEW',
    'module': 'Tracking Node',
    'createdAt': firestore.SERVER_TIMESTAMP,
    'userId': 'admin_test'
})

print("Injected fake error!")
